genum/internal/enumerable_with_parsable_traits.go

```diff
@@ -0,0 +1,12 @@
+//nolint:revive // test only
+package internal
+
+//go:generate genum -types=EnumerableWithParsableTraits -parsableByTraits=Parsable,AlsoSometimesNonParsable,OtherEnum
+
+type EnumerableWithParsableTraits int
```

Comment:
I modified this because it's kind of invalid (duplicate cases) for SometimesNon and AlsoSometimesNon... I'll paste it here because it was useful to find a few bugs. 

```go

//go:generate genum -types=EnumerableWithParsableTraits -parsableByTraits=Parsable1,Parsable2,OtherEnum,TypedString

type EnumerableWithParsableTraits int

const (
	P1, _NonParsable, _Parsable1, _Parsable2, _OtherEnum, _TypedString = EnumerableWithParsableTraits(iota), "non-parsable", 1, "1", Enum1Value0, OtherType("typedStr1")
	P2, _, _, _, _, _                                                  = EnumerableWithParsableTraits(iota), "non-parsable", 2, "2", Enum1Value1, OtherType("typedStr2")
	P3, _, _, _, _, _                                                  = EnumerableWithParsableTraits(iota), "non-parsable", 3, "3", Enum1Value2, OtherType("typedStr3")
)

```

Added to test file
```go


func TestEnumerableWithParsableTraits_MarshalJSON(t *testing.T) {
	for _, test := range internal.EnumerableWithParsableTraits.Values(0) {
		t.Run("json unmarshal "+test.String(), func(t *testing.T) {
			marshalInput := []byte(`"` + string(test.TypedString()) + `"`)
			result := *new(internal.EnumerableWithParsableTraits)
			require.NoError(t, result.UnmarshalJSON(marshalInput))
			assert.Equal(t, test.String(), result.String())
		})
	}
}
```